examples/scale-line.js

```diff
@@ -4,12 +4,27 @@ import {defaults as defaultControls, ScaleLine} from '../src/ol/control.js';
 import TileLayer from '../src/ol/layer/Tile.js';
 import OSM from '../src/ol/source/OSM.js';
 
+let scaleType = 'scaleline';
+let scaleBarSteps = 4;
+let scaleBarText = true;
+let control;
 
-const scaleLineControl = new ScaleLine();
-
+function scaleControl() {
+  if (scaleType === 'scaleline') {
+    control = new ScaleLine();
+    return control;
+  }
```

Comment:
I think you'll also need to set the units here.  As you can see in the [hosted example](https://488-4723248-gh.circle-artifacts.com/0/examples/scale-line.html), when you change from line to bar (or back), the units aren't right if they have been changed.